reproject/utils.py

```diff
@@ -45,15 +51,23 @@ def parse_output_projection(output_projection, shape_out=None):
         wcs_out = output_projection
         if shape_out is None:
             raise ValueError("Need to specify shape when specifying output_projection as WCS object")
+
     elif isinstance(output_projection, six.string_types):
         hdu_list = fits.open(output_projection)
         shape_out = hdu_list[0].data.shape
         header = hdu_list[0].header
         wcs_out = WCS(header)
         hdu_list.close()
+
+    elif isinstance(output_projection, NDDataBase):
+        wcs_out = output_projection.wcs
+        shape_out = output_projection.data.shape
+        output_array = output_projection.data
```

Comment:
output_projection shouldn't be used to mean in-place modification of array so this last line should be removed - but you could make it so that output_array can take an NDData object?